imap_processing/spice/repoint.py

```diff
@@ -42,16 +75,27 @@ def get_repoint_data() -> pd.DataFrame:
               Derived from `repoint_end_sec_sclk` and `repoint_end_subsec_sclk`.
             * `repoint_end_utc`: UTC time of repoint maneuver end time.
             * `repoint_id`: Unique ID number of each repoint maneuver.
+
+    Raises
+    ------
+    ValueError if no path to a repoint-table has been set.
     """
-    repoint_data_filepath = os.getenv("REPOINT_DATA_FILEPATH")
-    if repoint_data_filepath is not None:
-        path_to_spin_file = Path(repoint_data_filepath)
-    else:
-        # Handle the case where the environment variable is not set
-        raise ValueError("REPOINT_DATA_FILEPATH environment variable is not set.")
+    if _repoint_table_path is None:
+        raise ValueError(
+            "No repoint-table path as been defined in repoint.py "
+            "module attribute repoint_table_path."
+        )
+
+    logger.debug(
+        f"Reading in the following repoint table file: {_repoint_table_path.name}"
+    )
+
+    cache_key = str(_repoint_table_path.resolve())
+    if cache_key in _repoint_df_cache:
+        logger.debug("Returning cached repointing data")
+        return _repoint_df_cache[cache_key]
```

Comment:
I don't think this will work equivalently to what you have here since there is some extra processing involved, but wanted to mention that sometimes I have made a separate function that can be cached internal to your external function so that I don't have to make a cache and can just leverage the functools implementation. Something like:

```python
def get_repoint_data():
    path = _from_global
    repoint_df = _load_repoint_data_with_cache(path)

@functools.cache
def _load_repoint_data_with_cache(path):
    return pd.read_csv(path)
```